internal/weldr/api.go

```diff
@@ -1436,6 +1436,36 @@ func (api *API) composeStatusHandler(writer http.ResponseWriter, request *http.R
 		}
 	}
 	composes := api.store.GetAllComposes()
+
+	q, err := url.ParseQuery(request.URL.RawQuery)
+	if err != nil {
+		errors := responseError{
+			ID:  "InvalidChars",
+			Msg: fmt.Sprintf("invalid query string: %v", err),
+		}
+		statusResponseError(writer, http.StatusBadRequest, errors)
+		return
+	}
+
+	filterBlueprint := q.Get("blueprint")
+	filterStatus := q.Get("status")
+	filterImageType := q.Get("type")
+
+	if filterBlueprint != "" || filterStatus != "" || filterImageType != "" {
+		for uuid, compose := range composes {
+			if filterBlueprint != "" && compose.Blueprint.Name != filterBlueprint {
+				delete(composes, uuid)
+			}
+			if filterStatus != "" && compose.ImageBuilds[0].QueueStatus.ToString() != filterStatus {
+				delete(composes, uuid)
+			}
+			imageType, found := compose.ImageBuilds[0].ImageType.ToCompatString()
+			if filterImageType != "" && found && imageType != filterImageType {
+				delete(composes, uuid)
+			}
```

Comment:
I would move the `...ToCompatString()` up and then use `else if`. Your version is not wrong, but it might feel weird calling `delete` many times.